src/Storages/Statistics/UniqStatistic.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <Common/Arena.h>
+#include <Storages/Statistics/Statistics.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+
+namespace DB
+{
+
+class UniqStatistic : public IStatistic
+{
+    std::unique_ptr<Arena> arena;
+    AggregateFunctionPtr uniq_collector;
+    AggregateDataPtr data;
+    UInt64 result;
+public:
+    explicit UniqStatistic(const StatisticDescription & stat_, DataTypePtr data_type) : IStatistic(stat_), result(0)
+    {
+        arena = std::make_unique<Arena>();
+        AggregateFunctionProperties property;
+        property.returns_default_when_only_null = true;
+        uniq_collector = AggregateFunctionFactory::instance().get("uniq", NullsAction::IGNORE_NULLS, {data_type}, Array(), property);
+        data = arena->alignedAlloc(uniq_collector->sizeOfData(), uniq_collector->alignOfData());
+        uniq_collector->create(data);
+    }
+
+    ~UniqStatistic() override
+    {
+        uniq_collector->destroy(data);
```

Comment:
Hm, wondering why one needs to call `destroy` on a shared pointer to an aggregation function. Can't we just let it go out-of-scope?

Reply:
data is a `char *` and is not owned by aggFunction. if we do not explicitly release the aggregation data, it will leak.